### PR TITLE
Grand Maintenance Area Turfing (No More Secondary Edition)

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

This PR is a lot. I'm going to summarize the key points, and am happy to elaborate upon request.

In areas.dmi, we have a lot of obtuse "secondary" sprites, as well as pathing to that in game code.

image

Like that. It's... very not-helpful. "Secondary" as a term should have very limited use cases, as it's present usage in mapping tends to just be the stuff you chuck when you just don't want to areas to match. This PR is a delicate _shove_ away from that practice, and it does so by modernizing how maintenance tunnels are defined in their location to the station.

image

Ta-da! This PR adds a new "greater" and "lesser" sprite (and pathing) to the code. In-game and in-mapping, we do lay it all out on that 2-Dimensional plane, so greater refers to "above" and lesser refers to "below" (NOT their worth!). There are also "Central" variants of other directionalities for further clarification, as well as "Upper" for multi-z utilization. A lot of these were already included in maps, they just didn't have spiriting to match.

Since I decided to scrap /area/maintenance/*/secondary as a whole, I also remapped maintenance tunnels on probably every station currently in rotation to have better logic in their own naming. Some of the maps still have their edge-casey maps, but those will get addressed at a later date (because it's nightmarish in scope). They should all have their own APCs.

I also got rid of a few sprites that were not used anywhere in pathing, probably remnants of older maps? We should be fine. This should finally complete a goal I've had to get rid of numbers (specifically, the number "2") in area turfs, and I'm happy to see it fruit. Also, changed up names to be more verbose because it's harder to fuck up when shit's verbose.

I loaded up all the maps and SDMM screamed at me enough times so I'm pretty sure I caught all the areas. Hopefully? I'll check again when I have fresher eyes.

## Why It's Good For The Game

It makes more sense for mappers when they map, and it makes more sense for players interacting with the station (new names for maintenance tunnels). I like it, and I hope you do too.

Here's a fast example of what I'm talking about. This is what a snippet TramStation's maints currently looks like to me:

image

This is what I did to fix it:

image

Don't you think that makes more sense? (also, this touches a lot more maps than what's pictured)

## Changelog

:cl:
balance: Nanotrasen has updated how they name and define different parts of their loot-laden maintenance tunnels. Take a look around you!
/:cl:

### CODE REVIEW

**Comment by GoldenAlpharex** [comment]
I'll be real with you here. I don't map much if at all, I only occasionally review map PRs.

I absolutely don't understand what the fuck this aims to do.
Furthermore... What's the benefit of having Greater and Lesser maints, as well as why do we need them versus Upper and Lower maints (which at least make sense due to z-levels being involved).

I'm commenting this because in all honesty, from an outsider's perspective, this doesn't seem to actually make it any clearer for someone that's new to mapping, or new to messing with areas. Maybe it's just the PR body not explaining it too much, maybe I'm just very tired. But I 100% don't understand lmao.

**Comment by san7890** [comment]
>What's the benefit of having Greater and Lesser maints, as well as why do we need them versus Upper and Lower maints (which at least make sense due to z-levels being involved).

Because beforehand, you would have massive sprawling maint sections divided only be an arbitrary "secondary". For example, in the screenshot of tramstation above, you have two seperate sections of Central Maintenance. However, even though they are on polar ends of the station and are not contiguous in any way, they're both named "Central Maintenance". They have different APCs, yet to the player, they both show as "Central Maintenance". On the mapping end, they both have the sprite label "Central Maintenance", even though one is a "secondary" central maintenance.

More verbosity gives the player a better insight onto where stuff might be located on the station, rather than have two things that are very differently located named the same way.

**Comment by GoldenAlpharex** [comment]
I guess this makes more sense.

**Comment by san7890** [comment]
> I don't understand why you'd use greater and lesser and not fore and aft or bow and whatever we use instead of stern. Or even north and south. Greater and lesser makes it sound like one is smaller than the other, which is going to mislead mappers on how they're supposed to be used.

We already use port, aft, starboard, and fore. We also already have Aft-by-Starboard and Fore-by-Starboard for some maps. However, you still have massive Aft Maintenance Tunnels and gargantuan Fore Maintenance Tunnels with all the sub-divisions in the word. We don't, and shouldn't use north/south/east/west since four out of five current maps are on a space station.

Areas don't _exactly_ guide how maps are made. They might cause someone to shrink it, but I don't think there's a wrong way to have a map be fully drawn out and made and then settle on naming your maint tunnels. I wrote a small comment in the typepaths for all of the tunnels and how they should be used. This is already a much better step than "Aft Maintenance" and "Aft Maintenance Two". I was thinking of superior and inferior, but I think that might bleed through to the players and some might think "Superior" refers to the quality of the loot.

If they use greater for larger spaces and lesser for smaller spaces, that's fine. I'm not in a position to nanny or babysit mappers through how they wish to map and then label their maps. I just don't want to see area/secondary be used when they're on polar ends, or just two massive sections tangentially sequential with the same name and different APCs: "Port Maintenance"

**Review by Twaticus — APPROVED** [review]

**Comment by san7890** [comment]
have to go through and re-area for the modular tramstation maintenance. it has to be done.

File: code/game/area/space_station_13_areas.dm [hunk 1]
```diff
@@ -255,68 +255,91 @@ NOTE: there are two lists of areas in the end of this file: centcom and station
...
+	icon_state = "upperforemaint"
 
-/area/maintenance/fore/secondary
-	name = "Fore Maintenance"
-	icon_state = "fmaint_2"
+/area/maintenance/fore/greater
+	name = "Greater Fore Maintenance"
+	icon_state = "greaterforemaint"
+
+/area/maintenace/fore/lesser
```
Comment (Maurukas) [inline code/game/area/space_station_13_areas.dm:296]: Believe this should be /area/maintenance/fore/lesser
image

Comment (san7890) [reply code/game/area/space_station_13_areas.dm]: fixed the typo

**Pull request merged**